Library/Homebrew/test/cmd/install_spec.rb

```diff
@@ -72,4 +72,15 @@ def install
       .and be_a_success
     expect(HOMEBREW_CELLAR/"testball1/HEAD-d5eb689/foo/test").not_to be_a_file
   end
+
+  it "installs formulae with debug symbols", :integration_test do
+    setup_test_formula "testball1"
+
+    expect { brew "install", "testball1", "--debug-symbols", "--build-from-source" }
+      .to output(%r{#{HOMEBREW_CELLAR}/testball1/0\.1}o).to_stdout
+      .and not_to_output.to_stderr
+      .and be_a_success
+    expect(HOMEBREW_CELLAR/"testball1/0.1/bin/test").to be_a_file
+    expect(HOMEBREW_CELLAR/"testball1/0.1/bin/test.dSYM").to be_a_directory if OS.mac?
```

Comment:
What files are present on Linux? Would be good to verify it's doing something there. Would also be good to verify the expected files remain present in `HOMEBREW_CACHE`.

Reply:
> Would also be good to verify the expected files remain present in HOMEBREW_CACHE.

Done

> What files are present on Linux? Would be good to verify it's doing something there.

For linux the test appears to be to run something like:

```
objdump -h $HOMEBREW_CELLAR/"testball1/0.1/bin/test" | grep .debug
```

And check for any output.

I'm not in a position to test this, or even if the command is cross linux platform safe. But if you can test it, it would be great to add. Here's the reference: https://stackoverflow.com/questions/17453270/how-to-know-if-a-binary-contains-debugging-symbols-or-not-without-file-objdump

Reply:
I'm okay with relying on `objdump` being there. Not sure about the right way to add it to the test, though. It may even be possible to avoid using `objdump` and rely on Ruby `binread` directly, since we're just looking for a string in the binary.

However, I don't think `testball` actually compiles anything, so I wouldn't expect `objdump` to actually generate debug information.

Reply:
> rely on Ruby binread directly, since we're just looking for a string in the binary.

This would be ideal 👍🏻 

> However, I don't think testball actually compiles anything, so I wouldn't expect objdump to actually generate debug information.

I think it does compile a tiny C file or we won't have `test.dSYM`, right? If not: we can consider adjusting this test to compiled a tiny C file.

Regardless: worth adding to this PR and seeing if it passes or fails. If it fails: we can remove it and add a TODO to add something more involved later.

Reply:
> I think it does compile a tiny C file or we won't have `test.dSYM`, right? If not: we can consider adjusting this test to compiled a tiny C file.

It installs pre-built (32-bit!) binary objects:

https://github.com/Homebrew/brew/blob/8fea5ed63f80f771f8cbe6c89472cde49d27ebc6/Library/Homebrew/test/support/fixtures/testball.rb#L14-L18

Reply:
I do think it is compiling here. Check this code (which is called from the integration test): https://github.com/Homebrew/brew/blob/8fea5ed63f80f771f8cbe6c89472cde49d27ebc6/Library/Homebrew/test/support/helper/spec/shared_context/integration_test.rb#L136